### ISSUE TITLE
weedle 4.0.0 breaks uniffi_udl v0.25.3

### ISSUE THREAD

**Issue by simonsso** [mozilla/uniffi-rs#1957]
Just got the build error from a minor upgrade `4.0.0` to `4.0.1`  did not expect breaking changes in a minor release.

```
   Compiling uniffi_udl v0.25.3
error[E0609]: no field `0` on type `&EnumVariant<'_>`
  --> /home/simson/.cargo/registry/src/index.crates.io-6f17d22bba15001f/uniffi_udl-0.25.3/src/converters/enum_.rs:26:33
   |
26 |                         name: v.0.to_string(),
   |                                 ^ unknown field
   |
   = note: available fields are: `docstring`, `value`

error[E0609]: no field `0` on type `&EnumVariant<'_>`
  --> /home/simson/.cargo/registry/src/index.crates.io-6f17d22bba15001f/uniffi_udl-0.25.3/src/converters/enum_.rs:48:37
   |
48 | ...                   name: v.0.to_string(),
   |                               ^ unknown field
   |
   = note: available fields are: `docstring`, `value`

For more information about this error, try `rustc --explain E0609`.
error: could not compile `uniffi_udl` (lib) due to 2 previous errors
```

**Fixes:**
- Pining the version to `4.0.0` solves the problem
- Upgrade to `uniffi = { version = "0.26.0"}` also removed the problem

What is the real problem here? There is a hidden dependency to weedle2 where cargo assume it can upgrade to 4.0.1 wherein it cannot do so.

**Comment by skhamis** [comment]
Thanks for reporting this! You're correct that this has to do with the recent weedle bump to `4.0.1` which cargo toml will take even if `weedle2 = 4.0.1` as per https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio

@bendk we'll probably either need to patch 0.25.3 or majorly bump weedle2? (5.0.0) instead of patch?

**Comment by bendk** [comment]
Sorry about this.  The 0.26.0 release required a weedle update, which I thought was non-breaking but I was wrong about that.

Here's my plan to fix:
- `cargo yank` `weedle2 4.0.1`.  I think this should fix crates that depend on Uniffi `0.25.x`, although if you already have `weedle 4.01` in your `Cargo.lock` you'll need to do something to fix that.  I'm not exactly sure how that works, but I'll get myself in that situation locally and figure out the steps to fix things.
- Release `weedle2 5.0.0` and UniFFI `0.26.1` to fix the `0.26.x` branch.  Users of `0.26.0` will need to `cargo update` to prevent

**Comment by bendk** [comment]
'weedle2 4.0.1` is officially yanked.

If `4.0.1` is in your `Cargo.lock` file, you may need to run `cargo update weedle2` to downgrade back to `4.0.0`
